server/sonar-server/src/main/java/org/sonar/server/computation/step/ComputeQProfileMeasureStep.java

```diff
@@ -38,33 +38,36 @@
 import static org.sonar.server.computation.component.ComponentVisitor.Order.POST_ORDER;
 
 /**
- * Aggregates quality profile on lower-level module nodes on their parent modules and project
+ * Compute quality profile measure per module  based on present languages
  */
 public class ComputeQProfileMeasureStep implements ComputationStep {
 
   private final TreeRootHolder treeRootHolder;
   private final MeasureRepository measureRepository;
   private final MetricRepository metricRepository;
+  private final AnalysisMetadataHolder analysisMetadataHolder;
 
-  public ComputeQProfileMeasureStep(TreeRootHolder treeRootHolder, MeasureRepository measureRepository, MetricRepository metricRepository) {
+  public ComputeQProfileMeasureStep(TreeRootHolder treeRootHolder, MeasureRepository measureRepository, MetricRepository metricRepository,
+    AnalysisMetadataHolder analysisMetadataHolder) {
     this.treeRootHolder = treeRootHolder;
     this.measureRepository = measureRepository;
     this.metricRepository = metricRepository;
+    this.analysisMetadataHolder = analysisMetadataHolder;
   }
 
   @Override
   public void execute() {
     Metric qProfilesMetric = metricRepository.getByKey(CoreMetrics.QUALITY_PROFILES_KEY);
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Remove this use of "QUALITY_PROFILES_KEY"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)